plugins/mac-capture/mac-sck-audio-capture.m

```diff
@@ -0,0 +1,385 @@
+}
+
+static bool init_audio_screen_stream(struct screen_capture *sc)
+{
+	SCContentFilter *content_filter;
+
+	sc->stream_properties = [[SCStreamConfiguration alloc] init];
+	os_sem_wait(sc->shareable_content_available);
+
+	SCDisplay * (^get_target_display)() = ^SCDisplay *()
+	{
+		__block SCDisplay *target_display = nil;
+		[sc->shareable_content.displays
+			indexOfObjectPassingTest:^BOOL(
+				SCDisplay *_Nonnull display, NSUInteger idx,
+				BOOL *_Nonnull stop) {
+				if (display.displayID == sc->display) {
+					target_display = sc->shareable_content
+								 .displays[idx];
+					*stop = TRUE;
+				}
+				return *stop;
+			}];
+		return target_display;
+	};
+
+	switch (sc->capture_type) {
+	case ScreenCaptureAudioDesktopStream: {
+		SCDisplay *target_display = get_target_display();
+
+		content_filter = [[SCContentFilter alloc]
+			 initWithDisplay:target_display
+			excludingWindows:[[NSArray alloc] init]];
+	} break;
+	case ScreenCaptureAudioApplicationStream: {
+		SCDisplay *target_display = get_target_display();
+		__block SCRunningApplication *target_application = nil;
+		{
+			[sc->shareable_content.applications
+				indexOfObjectPassingTest:^BOOL(
+					SCRunningApplication
+						*_Nonnull application,
+					NSUInteger idx, BOOL *_Nonnull stop) {
+					if ([application.bundleIdentifier
+						    isEqualToString:
+							    sc->
+							    application_id]) {
+						target_application =
+							sc->shareable_content
+								.applications
+									[idx];
+						*stop = TRUE;
+					}
+					return *stop;
+				}];
+		}
+		NSArray *target_application_array = [[NSArray alloc]
+			initWithObjects:target_application, nil];
+
+		content_filter = [[SCContentFilter alloc]
+			      initWithDisplay:target_display
+			includingApplications:target_application_array
+			     exceptingWindows:[[NSArray alloc] init]];
+	} break;
+	}
+	os_sem_post(sc->shareable_content_available);
+	[sc->stream_properties setQueueDepth:8];
+
+	[sc->stream_properties setCapturesAudio:TRUE];
+	[sc->stream_properties setExcludesCurrentProcessAudio:TRUE];
+
+	struct obs_audio_info audio_info;
+	BOOL did_get_audio_info = obs_get_audio_info(&audio_info);
+	if (!did_get_audio_info) {
+		MACCAP_ERR(
+			"init_audio_screen_stream: No audio configured, returning %d\n",
+			did_get_audio_info);
+		return did_get_audio_info;
+	}
+	int channel_count = get_audio_channels(audio_info.speakers);
+	if (channel_count == 1) {
+		[sc->stream_properties setChannelCount:1];
+	} else {
+		[sc->stream_properties setChannelCount:2];
+	}
+
+	sc->disp = [[SCStream alloc] initWithFilter:content_filter
+				      configuration:sc->stream_properties
+					   delegate:nil];
+
+	//add a dummy video stream output to silence errors from SCK. frames are dropped by the delegate
+	NSError *error = nil;
+	BOOL did_add_output = [sc->disp addStreamOutput:sc->capture_delegate
+						   type:SCStreamOutputTypeScreen
+				     sampleHandlerQueue:nil
+						  error:&error];
+	if (!did_add_output) {
```

Comment:
If possible release `content_filter` after it's been passed to `initWithFilter` as it leaks otherwise.

Reply:
Fixed to release ``content_filter`` immediately after the ``initWithFilter`` call.